src/gui/XournalView.cpp

```diff
@@ -699,6 +699,9 @@ void XournalView::documentChanged(DocumentChangeType type) {
     }
     viewPages.clear();
 
+    delete this->cache;
+    this->cache = new PdfCache(control->getSettings()->getPdfPageCacheSize());
```

Comment:
This works, but I think it would be better to make `PdfCache::clearCache` public and call it instead, as it makes the intent clearer.